### PR TITLE
Feature/eks addon profile

### PR DESCRIPTION
**Pull request by jwlai**
Need some feedback for this chart. It installs properly but there are some items that I'm not too clear about.

### CODE REVIEW

**Comment by steve-fraser** [comment]
The autoscaler currently is configured differently based upon MachinePool and MachineDeplyoment respectively. In, the past CAPI basically fought the autoscaler group trying to reconcile the MachinePool replicas (https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2022) . I would consider keeping that out of this chart, but here is a few I would put in there:
(aws-node-termination-handler) https://github.com/filecoin-project/gitops-profile-catalog/tree/main/charts/aws-node-term-handler
(aws-ebs-csi-driver) https://github.com/filecoin-project/gitops-profile-catalog/tree/main/charts/aws-ebs-csi-driver
(aws-load-balancer-controller) https://github.com/filecoin-project/gitops-profile-catalog/tree/main/charts/aws-load-balancer-controller

**Comment by jwlai** [comment]
Since we're attaching multiple charts in one "profile", the chart.yaml annotation is fit for this design. Is this a question for engineering? (Kevin?)

i.e. One of the charts Steve mentioned has the "Operator" annotation, others have different "Upstream Project" Urls.

**Comment by darrylweaver** [comment]
aws-node-termination-handler would not be needed for an AWSManagedMachinePool see:
https://cluster-api-aws.sigs.k8s.io/topics/machinepools.html#awsmanagedmachinepool
and
https://github.com/aws/aws-node-termination-handler#project-summary
see the end of the project summary:

> You can run the termination handler on any Kubernetes cluster running on AWS, including self-managed clusters and those created with Amazon Elastic Kubernetes Service. If you're using EKS managed node groups, you don't need the aws-node-termination-handler.

So we would need to be able to turn it on or off, depending on the EKS cluster's objects in the template.

I would suggest we have a global parameter for turning on and off each of the subcharts.
An example would be the answer to this question showing how they do it for elk-stack charts:
https://stackoverflow.com/questions/55338147/disabling-subcharts-in-custom-helm-chart

Then we can include or exclude the subcharts with a simple parameter at the top level of values.yaml.

The issue with multiple sources.  Maybe it makes sense to have unique sources.  It may make sense to put the storage driver into it's own chart as it is a different source.
Then bundle the add-ons from the eks-charts repo together into one chart using the conditions to turn them on or off.

File: charts/karpenter/values.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  # Prerequisites
+  # See documentation for more instructions
+  # https://karpenter.sh/v0.16.1/getting-started/getting-started-with-eksctl/#install
+  clusterName: CLUSTER_NAME
+  clusterEndpoint: CLUSTER_ENDPOINT
+  aws:
+    defaultInstanceProfile: KarpenterNodeInstanceProfile-${CLUSTER_NAME}
+  serviceAccount:
+    annotations:
+      eks.amazonaws.com/role-arn: 'arn:aws:iam::123456789012:role/MyRoleName'
```
Comment (darrylweaver) [inline charts/karpenter/values.yaml:12]: Can this also be commented out?

**Review by darrylweaver — CHANGES_REQUESTED** [review]
Can the example ARN values be commented out?

We are not testing on EKS, which should be added ideally, but we could defer that to another Asana task.

**Review by darrylweaver — APPROVED** [review]

**Pull request merged**